### ISSUE TITLE
NumberInput does not allow entering decimal numbers when precision > 0

### ISSUE THREAD

**Issue by arel** [mantinedev/mantine#4505]
### What package has an issue

@mantine/core

### Describe the bug

On iOS (but not Android) it is impossible to enter a decimal number on a NumberInput when only the `precision` is set and not the `step`. The documentation for `precision` also does not say that `step` must be set in conjunction.

For instance, I want a NumberInput with two digits of precision, but I don't want the step functionality, so I have something like this:

```jsx
<NumberInput
  min={0}
  max={100}
  precision={2}
  hideControls
  placeholder="Enter percentage"
/>
```

Based on the logic in `get-input-mode`, the input mode will be set to `numeric`, which on iOS does not include the decimal key.

https://github.com/mantinedev/mantine/blob/d03cb692eedd7e525275bdedcc9b077b141f0b78/src/mantine-core/src/NumberInput/get-input-mode/get-input-mode.ts#L9

The solution would seem to be to change `get-input-mode` so that if `precision !== 0` the input mode is `decimal` (regardless of the value of `step`).

### What version of @mantine/hooks page do you have in package.json?

4.2.4 -- though I believe the problem still exists in the current version

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

In the `get-input-mode.ts` file, handle the case where `precison !== 0` and `step` is unset (or set to the default value).

In this case, the input mode should be `decimal` (or `text` on iOS, if negative numbers are to be expected.)

(As an aside, it seems that whether the mode is `decimal` or `text` currently depends on `step`, but shouldn't it depend on the `min` and `max` values of the NumberInput?)

**Comment by amin-hrp** [comment]
To take this a step further, decimal should be allowed even without precision, at least as an option.

**Comment by rtivital** [comment]
This issue is closed for one of these reasons:

- It was fixed in 7.0
- It is no longer reproducible in 7.0
- It is no longer applicable in 7.0
- It does not have a reproduction

If you think that this issue was closed by mistake and it is still an issue in version 7.0, please reopen it.
